### PR TITLE
Support override vcpkg installed dir

### PR DESCRIPTION
**Pull request by Danielmelody**
This might be more generic than #42 . As manifest mode is a sub case of overriding an existing vcpkg installed directory, So this pr is made.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -721,12 +727,13 @@ fn load_ports(target: &VcpkgTarget) -> Result<BTreeMap<String, Port>, Error> {
     // load updates to the status file that have yet to be normalized
     let status_update_dir = target.status_path.join("updates");
 
-    let paths = try!(
-        fs::read_dir(status_update_dir).map_err(|e| Error::VcpkgInstallation(format!(
-            "could not read status file updates dir: {}",
+    let paths = try!(fs::read_dir(status_update_dir.clone()).map_err(
```
Comment (pkgw) [inline src/lib.rs:730]: For what it's worth, I think it should work to use `&status_update_dir` here and save the clone.

**Comment by pkgw** [comment]
Thank you for your submission! This looks reasonable to me in isolation, although I don't feel like I understand the broader context of when this mode is useful. (Which is just a matter of my ignorance, not a problem with the PR.)

File: src/lib.rs [hunk 2]
```diff
@@ -1056,6 +1063,12 @@ impl Config {
         self
     }
 
+    /// Specify vcpkg installed directory. This is useful for manifest mode and custom vcpkg installation.
```
Comment (pkgw) [inline src/lib.rs:1066]: A little bit more explanation here would be nice, as well as a mention that the default is `$VCPKG_ROOT/installed`.

**Comment by Danielmelody** [comment]
In a hybrid C++/Rust project, one can use a vcpkg.json manifest file to manage vcpkg installations. The path for vcpkg installations can be altered to any desired location using CMake. 'vcpkg-rs' thus can be a bridge between the existing vcpkg installation and Rust, even in manifest mode.

File: src/lib.rs (hunk 2 above)
Comment (Danielmelody) [reply src/lib.rs:1066]: Added below

**Comment by pkgw** [comment]
@waych Any feedback on this one?

**Comment by waych** [comment]
Hmm. My initial thought is that this is fine and can go in as is.

Thinking longer term, I'm not particularly fond of the way this extends the interface on the Config directly, because this can cause problems with dependent crates that also want to pull in vcpkg dependencies.  This makes me think that perhaps we should be propagating this information down through the crate dependencies somehow so that dependent crates' vcpkg dependencies can be fulfilled by a single top level vcpkg.json manifest. Given that there isn't any good way for the top-level crate to pass this information to dependent crates (?) from build.rs, it seems perhaps it would be clearer, although maybe more inconvenient to be passing this installed path via an environment variable where all vcpkg-rs::Config's in the build tree see the same thing.

That said I think this is okay for now given the problem already exists in the config interface with the existing `Config::vcpkg_root()` setter that has the exact same problem. Any library crate using these will have similar challenges. This kiss approach is good enough for now and can easily be further extended with an environment variable when the time comes.

**Comment by pkgw** [comment]
Hmmm. I can see how environment variables can address the issue you raise, although I guess it's less "passing information from the top-level crate to dependents" and more "user specifies global-level information".

Maybe there would be some way to specify this configuration, or where to find this configuration, in the Cargo.toml metadata? I am wondering if `build.rs` scripts can use some magic to get information about the toplevel Cargo.toml in their workspace, not just their local file.

**Comment by Danielmelody** [comment]
Thank you for those good reviews. I think perhaps adding environment variable like `VCPKG_INSTALLED_ROOT` should workaround the problem. As we already have `VCPKG_ROOT` playing the same role.  One thing need to concern is that `VCPKG_ROOT` is also a env used by vcpkg itself, while `VCPKG_INSTALLED_ROOT` is a env defined by ourselves.
Considering a real example, [ffmpeg-next-sys](https://crates.io/crates/ffmpeg-sys-next) tries to using vcpkg in their [build.rs](https://github.com/zmwangx/rust-ffmpeg-sys/blob/e14770de907e310ccd62c8b0e8855feb85ebb8f1/build.rs#L393).  
Defining environment `VCPKG_INSTALLED_ROOT` from the top crate's `config.toml` is the only way to pass vcpkg installed location information to `ffmpeg-next-sys`. So I guess it is valuable to extend environment variable at this moment.

**Comment by pkgw** [comment]
OK; more to do on this topic, but I've merged this piece.

**Pull request merged**